controllers/contentlibrary/v1alpha1/utils/utils.go

```diff
@@ -45,23 +47,68 @@ func IsItemReady(itemConditions imgregv1a1.Conditions) bool {
 	return isReady
 }
 
-// GetVMImageSpecStatus returns the VirtualMachineImage Spec and Status fields in an image-registry service enabled cluster.
-// It first tries to get the namespace scope VM Image; if not found, it looks up the cluster scope VM Image.
-func GetVMImageSpecStatus(ctx context.Context, ctrlClient client.Client, imageName, namespace string) (
-	spec *vmopv1.VirtualMachineImageSpec, status *vmopv1.VirtualMachineImageStatus, err error) {
-
+// GetVMImageSpecStatus retrieves the VirtualMachineImageSpec and
+// VirtualMachineImageStatus for a given image name and namespace.
+// It first checks if a namespace scope image exists by the resource name,
+// and if not, checks if a namespace scope image exists by the status name.
+// If neither is found, it checks a cluster scope image with the same order.
+func GetVMImageSpecStatus(
+	ctx context.Context,
+	ctrlClient client.Client,
+	imageName, namespace string) (
+	*vmopv1.VirtualMachineImageSpec,
+	*vmopv1.VirtualMachineImageStatus,
+	error) {
+	// Check if a namespace scope image exists by the resource name.
 	vmi := vmopv1.VirtualMachineImage{}
-	if err = ctrlClient.Get(ctx, client.ObjectKey{Name: imageName, Namespace: namespace}, &vmi); err == nil {
-		spec = &vmi.Spec
-		status = &vmi.Status
-	} else if apierrors.IsNotFound(err) {
-		// Namespace scope image is not found. Check if a cluster scope image with this name exists.
-		cvmi := vmopv1.ClusterVirtualMachineImage{}
-		if err = ctrlClient.Get(ctx, client.ObjectKey{Name: imageName}, &cvmi); err == nil {
-			spec = &cvmi.Spec
-			status = &cvmi.Status
+	key := client.ObjectKey{Name: imageName, Namespace: namespace}
+	err := ctrlClient.Get(ctx, key, &vmi)
+	if err == nil {
+		return &vmi.Spec, &vmi.Status, nil
+	}
+	if !apierrors.IsNotFound(err) {
+		return nil, nil, err
+	}
+
+	// Check if a namespace scope image exists by the image status name.
+	vmiList := vmopv1.VirtualMachineImageList{}
+	err = ctrlClient.List(ctx, &vmiList, client.InNamespace(namespace))
+	if err == nil {
+		for _, vmi := range vmiList.Items {
+			if vmi.Status.ImageName == imageName {
```

Comment:
I think we need to add a check to see this vmi is the only one with this image name. Because for one image that is uploaded to multiple content library in the same ns, we cannot just return the first one found.

Reply:
Good point. I was trying to best effort the image lookup process earlier. After further reflection, I think it would be better to throw an error if duplicate image status names exist as you suggested. This avoids unexpected behaviors, especially if those images are significantly different, but happen to have the same status name.